src/data/moves/move.ts

```diff
@@ -2711,46 +2726,69 @@ export class EatBerryAttr extends MoveEffectAttr {
     globalScene.updateModifiers(target.isPlayer());
   }
 
-  eatBerry(consumer: Pokemon, berryOwner?: Pokemon) {
-    getBerryEffectFunc(this.chosenBerry!.berryType)(consumer, berryOwner); // consumer eats the berry
+
+  /**
+   * Internal function to apply berry effects.
+   *
+   * @param consumer - The {@linkcode Pokemon} eating the berry; assumed to also be owner if `berryOwner` is omitted
+   * @param berryOwner - The {@linkcode Pokemon} whose berry is being eaten; defaults to `consumer` if not specified.
+   * @param updateHarvest - Whether to prevent harvest from tracking berries;
+   * defaults to whether `consumer` equals `berryOwner` (i.e. consuming own berry).
+   */
+   eatBerry(consumer: Pokemon, berryOwner: Pokemon = consumer, updateHarvest = consumer === berryOwner) {
```

Comment:
If this is an internal function then it should be `protected` or `private`
```suggestion
   protected eatBerry(consumer: Pokemon, berryOwner: Pokemon = consumer, updateHarvest = consumer === berryOwner) {
```